packages/docusaurus/bin/beforeCli.mjs

```diff
@@ -26,6 +27,29 @@ const {
   engines: {node: requiredVersion},
 } = packageJson;
 
+function checkDependencyVersions() {
+  // notify user if node version needs to be updated
+  if (!semver.satisfies(process.version, requiredVersion)) {
+    logger.error('Minimum Node.js version not met :(');
+    logger.info`You are using Node.js number=${process.version}, Requirement: Node.js number=${requiredVersion}.`;
+    process.exit(1);
+  }
+
+  // React 18 introduced new render APIs that don't exist in previous versions.
+  const reactDomVersion = ReactDOM.version;
+  const isReactExperimental = /0\.0\.0-experimental/.test(reactDomVersion);
+  const hasReact18 =
+    Boolean(reactDomVersion) &&
+    (semver.gte(reactDomVersion, '18.0.0') ||
+      semver.coerce(reactDomVersion)?.version === '18.0.0');
```

Comment:
I don't think Docusaurus users care much about using experimental React versions

Do we really need this? Is it inspired by another codebase?

Reply:
It's better that we can take care of every edge case. It's highly unlikely but adding a check doesn't hurt anyways